evalml/pipelines/components/transformers/encoders/onehot_encoder.py

```diff
@@ -180,10 +178,38 @@ def categories(self, feature_name):
             raise ValueError(f'Feature "{feature_name}" was not provided to one-hot encoder as a training feature')
         return self._encoder.categories_[index]
 
+    @staticmethod
+    def _make_name_unique(name, seen_before):
+        """Helper to make the name unique."""
+        i = 1
+        name = f"{name}_{i}"
+        while name in seen_before:
+            name = f"{name[:name.rindex('_')]}_{i}"
+            i += 1
+        return name
+
     def get_feature_names(self):
         """Return feature names for the input features after fitting.
 
         Returns:
             np.array: The feature names after encoding, provided in the same order as input_features.
         """
-        return self._encoder.get_feature_names(self.features_to_encode)
+        unique_names = []
+        seen_before = set([])
+        for col_index, col in enumerate(self.features_to_encode):
+            column_categories = self.categories(col)
+            for cat_index, category in enumerate(column_categories):
+
+                # Drop categories specified by the user
+                if self._encoder.drop_idx_ is not None and self._encoder.drop_idx_[col_index] is not None:
+                    if cat_index == self._encoder.drop_idx_[col_index]:
+                        continue
+
+                # Follow sklearn naming convention but if name has been seen before
+                # then add an int to make it unique
+                proposed_name = f"{col}_{category}"
+                if proposed_name in seen_before:
+                    proposed_name = self._make_name_unique(proposed_name, seen_before)
```

Comment:
Really nit-picky comment but we can keep the logic of making a unique name in `make_name_unique()` by not checking here, and doing so in the helper instead. What I mean is here, 
```
proposed_name = self._make_name_unique(f"{col}_{category}", seen_before)
```
And then in `_make_name_unique`:
```
    def _make_name_unique(name, seen_before):
        """Helper to make the name unique."""
        i = 1
        while name in seen_before:
            name = f"{name[:name.rindex('_')]}_{i}"
            i += 1
        return name
```

Something like that? :D

Reply:
Good suggestion! I had to add some minor modifications but the loop within `get_feature_names` is easier to read now.